feat(scopes): Add Distribution permission UI for integration tokens

runningcode opened the pull request:
This PR adds frontend support for the `project:distribution` permission scope, building on top of #102295.

## Changes

- Add `Distribution` resource to `SENTRY_APP_PERMISSIONS` with read-only access
- Add `project:distribution` to `API_ACCESS_SCOPES` and `ALLOWED_SCOPES`
- Update `Permissions` TypeScript type to include optional `Distribution` field
- Add scope conversion logic for `project:distribution` in `consolidatedScopes.tsx`
- Add `hiddenPermissions` prop to `PermissionSelection` component to filter out specific permissions
- Use `hiddenPermissions` in personal token creation UI to hide Distribution permission

## Behavior

**Integration Tokens (Sentry Apps):**
- Can select Distribution permission with read-only access
- Grants the `project:distribution` scope

**Personal Tokens:**
- Distribution permission is hidden from the UI
- Cannot access distribution-related endpoints

This ensures that only integration tokens can use the distribution permission, while personal tokens cannot.

## Depends On

- #102295 (backend changes)

runningcode committed:
feat(scopes): Add Distribution permission UI for integration tokens

Add frontend support for the project:distribution permission scope:
- Add Distribution resource to SENTRY_APP_PERMISSIONS with read-only access
- Update scope lists and TypeScript types
- Add scope conversion logic for project:distribution
- Hide Distribution permission from personal token creation UI
  by adding hiddenPermissions prop to PermissionSelection component

This allows integration tokens to be created with Distribution
permission while preventing personal tokens from accessing it.

chromy commented on static/app/types/integrations.tsx:
```diff
@@ -24,6 +24,7 @@ export type Permissions = {
   Team: PermissionValue;
   Alerts?: PermissionValue;
+  Distribution?: PermissionValue;
 };
```
How come only `Alerts` and `Distribution` are optional (with the `?`).

runningcode committed:
ref(scopes): Use positive displayedPermissions API for permission selection

Replace the negative hiddenPermissions prop with a positive
displayedPermissions prop in PermissionSelection component:

- Export PermissionObj type from constants for reuse
- Change prop from hiddenPermissions to displayedPermissions
- Update apiNewToken to filter displayed permissions positively
- Update Distribution help text to be more descriptive

This provides a clearer API where callers explicitly specify which
permissions to show rather than which to hide.

runningcode commented on static/app/constants/index.tsx:
```diff
@@ -121,7 +121,7 @@ type PermissionChoice = {
 };
 
-type PermissionObj = {
+export type PermissionObj = {
```
This needs to be exported to be used in permissionSelection.tsx

runningcode committed:
feat(scopes): Gate Distribution permission behind preprod feature flag

Only show the Distribution permission in personal token creation
when the organizations:preprod-frontend-routes feature flag is enabled.

This ensures the Distribution permission is only available to users
with access to preprod features.

runningcode committed:
fix(scopes): Use filteredScopes consistently in toResourcePermissions

The Release and Distribution scope filtering was incorrectly filtering
from the original scopes array instead of the progressively filtered
filteredScopes array, which could cause scopes to not be properly removed.

runningcode committed:
ref(scopes): Use preprod-build-distribution flag for Distribution permission

Switch from the generic preprod-frontend-routes flag to the more specific
preprod-build-distribution flag for gating the Distribution permission in
integration token creation.

runningcode committed:
fix(scopes): Add Distribution to default permissions state

Add Distribution permission to the default permissions state object
to ensure it's properly tracked in the integration token creation form.

chromy reviewed (approved):
lgtm
